### PR TITLE
Extended 'Active Job Basics' guide with more detailed information

### PR DESCRIPTION
**Pull request by ZeroMax-dev**
Hello,

I wanted to extend the guide on Active Job Basics a bit. The guide is still 'edge' and as I was working myself through it on the official page I had a harder time than usual setting everything up in my app because in my opinion there were some important information missing, especially for newbies that have not much experience.

The additions are small but I found it crucial to add them.

### CODE REVIEW

File: guides/source/active_job_basics.md [hunk 1]
```diff
@@ -123,6 +150,11 @@ module YourApp
 end
 ```
 
+NOTE: Since jobs run in parallel to your Rails application, most queuing implementations
+require that you start an implementation-specific queuing service (in addition to
+starting your Rails app) for the job processing to work. For information on
+how to do that refer to the documentation of your respective queuing implementation.
```
Comment (zzak) [inline guides/source/active_job_basics.md]: s/queuing implementation/queuing library.

I also never remember if it's queuing or queueing.. such as above, we have "enqueuing".

File: guides/source/active_job_basics.md [hunk 2]
```diff
@@ -123,6 +150,11 @@ module YourApp
...
+
```
Comment (zzak) [inline guides/source/active_job_basics.md]: This extra line should be :scissors:

File: guides/source/active_job_basics.md [hunk 3]
```diff
@@ -73,6 +78,25 @@ class GuestsCleanupJob < ActiveJob::Base
...
+Rails does not auto-load the `app/jobs` directory by default. To make your job
+accessible in the rest of your application you need to add the directory to the
+auto-load paths:
+
+```ruby
+# config/application.rb
+module YourApp
+  class Application < Rails::Application
+    # Lets Rails find and load your job classes for use in your code
```
Comment (zzak) [inline guides/source/active_job_basics.md]: s/Lets/Let ... "and autoload your job classes"...

File: guides/source/active_job_basics.md [hunk 4]
```diff
@@ -73,6 +78,25 @@ class GuestsCleanupJob < ActiveJob::Base
...
+    config.autoload_paths << Rails.root.join('app/jobs')
+  end
+end
+```
+
+You can then make use of the job anywhere in your code (e.g. in your controllers
```
Comment (zzak) [inline guides/source/active_job_basics.md]: "Now you can make use of"...

File: guides/source/active_job_basics.md [hunk 5]
```diff
@@ -28,11 +28,16 @@ into small units of work and run in parallel, really.
...
-framework features and other gems build on top of that, without having to
-worry about API differences between various job runners such as Delayed Job
-and Resque. Picking your queuing backend becomes more of an operational concern,
-then. And you'll be able to switch between them without having to rewrite your jobs.
+in place. We can then have framework features and other gems build on top of that,
+without having to worry about API differences between various job runners such as
+Delayed Job and Resque. Picking your queuing backend becomes more of an operational
+concern, then. And you'll be able to switch between them without having to rewrite
+your jobs.
+
```
Comment (zzak) [inline guides/source/active_job_basics.md]: Remove one of the newlines

File: guides/source/active_job_basics.md [hunk 6]
```diff
@@ -28,11 +28,16 @@ into small units of work and run in parallel, really.
...
+
+NOTE: Rails by default comes with an "immediate runner" queueing implementation.
+That means that each job that has been enqueued is run immediately without further
```
Comment (zzak) [inline guides/source/active_job_basics.md]: ... "has been enqueued will run immediately" ...

File: guides/source/active_job_basics.md [hunk 7]
```diff
@@ -28,11 +28,16 @@ into small units of work and run in parallel, really.
...
+options.
```
Comment (zzak) [inline guides/source/active_job_basics.md]: Can remove "without further options", please

File: guides/source/active_job_basics.md [hunk 8]
```diff
@@ -99,7 +123,10 @@ That's it!
 Job Execution
 -------------
 
-If no adapter is set, the job is immediately executed.
+For enqueuing and executing jobs you need to set up a queuing backend, that is to
+say you need to decide for a 3rd-party queuing implementation that Rails should use.
```
Comment (zzak) [inline guides/source/active_job_basics.md]: I don't think this is necessarily true, since the inline adapter will be used _if you don't choose_.

**Comment by zzak** [comment]
LGTM, just some minor nitpicks.

Thanks!

File: guides/source/active_job_basics.md (hunk 1 above)
Comment (ZeroMax-dev) [inline guides/source/active_job_basics.md]: I realized that the use of enqueueing/enqueuing has already been inconsistent throughout the document. I could normalize the other uses, too. I propose "enqueuing" as the stanard.

**Comment by zzak** [comment]
@ZeroMax-dev Can you squash your commits please?

**Comment by ZeroMax-dev** [comment]
Yep, done.

**Comment by ZeroMax-dev** [comment]
Yep, done.

**Comment by zzak** [comment]
@ZeroMax-dev Thank you!

**Pull request merged**